src/Scientist/Internals/Experiment.cs

```diff
@@ -18,15 +18,17 @@ internal class Experiment<T> : IExperiment<T>, IExperimentAsync<T>
         private readonly Dictionary<string, Func<Task<T>>> _candidates;
         private Func<T, T, bool> _comparison = DefaultComparison;
         private Func<Task> _beforeRun;
+        private readonly ILaboratory _laboratory;
         private Action<Operation, Exception> _thrown = _alwaysThrow;
         private Func<Task<bool>> _runIf = _alwaysRun;
         private readonly List<Func<T, T, Task<bool>>> _ignores = new List<Func<T, T, Task<bool>>>();
         private readonly Dictionary<string, dynamic> _contexts = new Dictionary<string, dynamic>();
 
-        public Experiment(string name)
+        public Experiment(string name, ILaboratory laboratory)
```

Comment:
I think my idea was to encapsulate other things into this interface at a global level, but that could easily be done as needed when additional things are identified.

What about implementing it like this?

```csharp
class Scientist {
    internal static Func<Task<bool>> _enabled = () => Task.FromResult(true);

    public static void Enabled(Func<bool> enabled) { ... }
    public static void Enabled(Func<Task<bool>> enabled) { ... }

    public static T Science<T>(...) {
        var experiment = new Experiment<T>(name, _enabled);
    }
}
```